artifactory/commands/transferfiles/errorshandler.go

```diff
@@ -0,0 +1,243 @@
+		return err
+	}
+
+	return nil
+}
+
+func (mng *TransferErrorsMng) newContentWriter(dirPath string, index int) (*content.ContentWriter, string, error) {
+	writer, err := content.NewContentWriter("errors", true, false)
+	if err != nil {
+		return nil, "", err
+	}
+	errorsFilePath := path.Join(dirPath, fmt.Sprintf("%s-%d-%s-%d.json", mng.repoName, mng.phaseId, mng.phaseStartTime, index))
+	return writer, errorsFilePath, nil
+}
+
+func (mng *TransferErrorsMng) writeErrorContent(e FileUploadStatusResponse) error {
+	switch e.Status {
+	case SkippedLargeProps:
+		log.Info(fmt.Sprintf("write %s to file %s", e.Reason, mng.errorWriterMng.skipped.filePath))
+		mng.errorWriterMng.skipped.writer.Write(e)
+		mng.errorWriterMng.skipped.errorCount++
+		// If file contains maximum number off errors - create and write to a new errors file
+		if mng.errorWriterMng.skipped.errorCount == maxErrorsInFile {
+			err := mng.errorWriterMng.skipped.closeWriter()
+			if err != nil {
+				return err
+			}
+			// Initialize variables for new errors file
+			mng.errorWriterMng.skipped.fileIndex++
+			dirPath, err := coreutils.GetJfrogTransferSkippedDir()
+			if err != nil {
+				return err
+			}
+			mng.errorWriterMng.skipped.writer, mng.errorWriterMng.skipped.filePath, err = mng.newContentWriter(dirPath, mng.errorWriterMng.skipped.fileIndex)
+			if err != nil {
+				return err
+			}
+			mng.errorWriterMng.skipped.errorCount = 0
+			return nil
+
+		}
+	default:
+		log.Info(fmt.Sprintf("write %s to file %s", e.Reason, mng.errorWriterMng.retryable.filePath))
+		mng.errorWriterMng.retryable.writer.Write(e)
+		mng.errorWriterMng.retryable.errorCount++
+		// If file contains maximum number off errors - create and write to a new errors file
+		if mng.errorWriterMng.retryable.errorCount == maxErrorsInFile {
+			err := mng.errorWriterMng.retryable.closeWriter()
+			if err != nil {
+				return err
+			}
+			// Initialize variables for new errors file
+			mng.errorWriterMng.retryable.fileIndex++
+			dirPath, err := coreutils.GetJfrogTransferRetryableDir()
+			if err != nil {
+				return err
+			}
+			mng.errorWriterMng.retryable.writer, mng.errorWriterMng.retryable.filePath, err = mng.newContentWriter(dirPath, mng.errorWriterMng.retryable.fileIndex)
+			if err != nil {
+				return err
+			}
+			mng.errorWriterMng.retryable.errorCount = 0
+			return nil
+		}
+	}
+	return nil
+}
+
+func (writerMng *errorWriter) closeWriter() error {
+	// Close content writer and move output file to our working directory
+	if writerMng.writer == nil {
+		return nil
+	}
+	err := writerMng.writer.Close()
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+
+	if writerMng.writer.GetFilePath() != "" {
+		log.Debug(fmt.Sprintf("Save errors outpt in: %s.", writerMng.filePath))
+		err = fileutils.MoveFile(writerMng.writer.GetFilePath(), writerMng.filePath)
+		if err != nil {
+			log.Error(fmt.Sprintf("Saving error file failed : failed to move %s to %s", writerMng.writer.GetFilePath(), writerMng.filePath))
```

Comment:
1. You should concat this error message to the message of the error returned - and not add this as a log message.
2. Also, the error message format is a bit strange with the " : " in the middle of the message. Let's phrase the error message properly.
3. You can also reduce the two return statement into one.